pkg/base/replicated.go

```diff
@@ -152,7 +156,54 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, kotsKindsFileMap, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+
+	for _, upstreamFile := range upstreamFiles {
+		gvk := OverlySimpleGVK{}
+		if err := yaml.Unmarshal(upstreamFile.Content, &gvk); err != nil {
+			continue
+		}
+		gvkString := gvk.APIVersion + "," + gvk.Kind
```

Comment:
the switch statement below includes a space, but this doesn't seem to. i would also suggest using `fmt`:
```suggestion
		gvkString := fmt.Sprintf("%s, %s", gvk.APIVersion, gvk.Kind)
```